Unit testing kmc model

meelod opened the pull request:
Replaced pt = Project and pt. with kmcos.create_kmc_model and kmc_model.

meelod committed:
Edit to test fext file

Added description on installing UnitTesterSG(somplete) and note to use pytest filename to test individual files. Made grammatical corrections as well.

meelod committed:
Delete 220201TestDirectoriesStatus.txt

meelod committed:
Merge branch 'kmcos:master' into test_txt_file

meelod committed:
Examples

Added kmcos.create_kmc_model and kmcos_compile to all example files (now listed and named as ...__build.py

meelod committed:
Examples

Removed local_smart directories that were not in previous kmcos example version

meelod committed:
Comment corrections

meelod committed:
Edits to text file

meelod committed:
Merge branch 'kmcos:master' into test_txt_file

AdityaSavara committed:
Update 220201TestDirectoriesStatus.txt

AdityaSavara committed:
Update 220202TestDirectoriesStatus.txt

AdityaSavara committed:
Delete 220201TestDirectoriesStatus.txt

AdityaSavara committed:
Merge pull request #141 from meelod/test_txt_file

Test txt file

meelod committed:
MyThirdTPR__build

meelod committed:
kmc_model improvements

Added save_model() with a save() pointer for backwards compatibility. Added assignments to Project object in set meta rather than just the meta.

meelod committed:
Merge branch 'kmcos:master' into create_kmc_model

meelod committed:
Updates to kmcos examples

Reflects standard commands at the bottom of all __build.py files

meelod committed:
Merge branch 'create_kmc_model' of https://github.com/meelod/kmcos into create_kmc_model

meelod committed:
Updates to saving and clearing model

meelod committed:
Revert "Updates to kmcos examples"

This reverts commit eb2ed7607eab568db0e4e7d17fa09c8c263515ca.

meelod committed:
Updates to clear model

meelod committed:
Removal of all xml files

AdityaSavara committed:
Merge pull request #144 from meelod/create_kmc_model

Examples

meelod committed:
Updates to test files

Includes create_kmc_model syntax

meelod committed:
create_kmc_model

meelod committed:
Added dynamic model_name based on file name

meelod committed:
Update __init__.py

meelod commented:
Do not merge yet: still need to change file names to __build. Currently facing errors after changing the names and still need to be addressed.

AdityaSavara commented on tests/accelerated_tests/CO_oxidation_Ruo2__acc_backend__do_acc_steps.py:
```diff
@@ -432,15 +433,10 @@
-import kmcos
+# Save the model to an xml file
 ###It's good to simply copy and paste the below lines between model creation files.
-pt.filename = model_name + ".xml"
```
@meelod it is not surprising that this test would break.  You have simply pasted over the code here, when the acceleration option is needed. The '-t' option is needed within the compile_options, and you have not (as far as I know) added the compile_options code.  So this example needs to continue using the "export" command.

AdityaSavara commented on tests/export_test/test_import_export.py:
```diff
@@ -206,26 +206,26 @@ def off_compare_import_variants():
     os.chdir(cwd)
 if __name__ == '__main__':
      test_import_export()
-     test_compare_import_variants()
```
@meelod , I don't understand why you have removed line 231 here. It does not make sense to me that you would remove a unit test line entirely.

meelod committed:
Update __init__.py

meelod committed:
Merge pull request #1 from kmcos/develop

Develop

meelod committed:
Updating version number in init

meelod committed:
acceleration tests

Current changes causes pytest to fail (different then before). Running pytest on the indivdual test files still works.

meelod commented:
Current changes in 'acceleration test' commit causes pytest to fail (different then before). Running pytest on the indivdual test files still works.

AdityaSavara commented:
> Current changes in 'acceleration test' commit causes pytest to fail (different then before). Running pytest on the indivdual test files still works.

Ok, please put a copy of the old version and new version in your local copy (like _old and _new) that way we can switch back and forth between them to show me.

meelod committed:
Update types.py

meelod committed:
Update throttling_test_reaction.py

meelod commented:
Issue fixed with SQERTSS. Ready to merge

meelod committed:
Update throttling_test_reaction.py

AdityaSavara reviewed (approved):
Meelod says all unit tests are behaving as expected

meelod committed:
Update test_import_export.py
